python/ray/train/batch_predictor.py

```diff
@@ -172,21 +173,21 @@ def calculate_accuracy(df):
 
         predictor_cls = self._predictor_cls
         checkpoint_ref = self._checkpoint_ref
-        predictor_kwargs = self._predictor_kwargs
+        predictor_kwargs_ref = self._predictor_kwargs_ref
         override_prep = self._override_preprocessor
         # Automatic set use_gpu in predictor constructor if user provided
         # explicit GPU resources
         if (
             "use_gpu" in inspect.signature(predictor_cls.from_checkpoint).parameters
-            and "use_gpu" not in predictor_kwargs
+            and "use_gpu" not in self._predictor_kwargs
             and num_gpus_per_worker > 0
         ):
             logger.info(
                 "`num_gpus_per_worker` is set for `BatchPreditor`."
                 "Automatically enabling GPU prediction for this predictor. To "
                 "disable set `use_gpu` to `False` in `BatchPredictor.predict`."
             )
-            predictor_kwargs["use_gpu"] = True
+        self._predictor_kwargs["use_gpu"] = True
```

Comment:
This won't make it into the kwargs in the object store, since we `ray.put()` before this (in the constructor). That's why I pushed the `ray.put()` to after this line in the diff I sent, so this `"use_gpu"` setting is included.